zipline/gens/brokers/rh/robinhood.py

```diff
@@ -0,0 +1,1147 @@
+            res.raise_for_status()
+            data = res.json()
+        except requests.exceptions.HTTPError:
+            raise RH_exception.LoginFailed()
+
+        if 'mfa_required' in data.keys():           #pragma: no cover
+            raise RH_exception.TwoFactorRequired()  #requires a second call to enable 2FA
+
+        if 'token' in data.keys():
+            self.auth_token = data['token']
+            self.headers['Authorization'] = 'Token ' + self.auth_token
+            return True
+
+        return False
+
+
+    def logout(self):
+        """Logout from Robinhood
+
+        Returns:
+            (:obj:`requests.request`) result from logout endpoint
+
+        """
+
+        try:
+            req = self.session.post(self.endpoints['logout'])
+            req.raise_for_status()
+        except requests.exceptions.HTTPError as err_msg:
+            warnings.warn('Failed to log out ' + repr(err_msg))
+
+        self.headers['Authorization'] = None
+        self.auth_token = None
+
+        return req
+
+
+    ###########################################################################
+    #                               GET DATA
+    ###########################################################################
+
+    def get_url_content_json(self, url):
+        """fetch url content"""
+        res = self.session.get(url)
+        res.raise_for_status()  #will throw without auth
+        data = res.json()
+        return data
+
+    def investment_profile(self):
+        """Fetch investment_profile """
+
+        res = self.session.get(self.endpoints['investment_profile'])
+        res.raise_for_status()  #will throw without auth
+        data = res.json()
+
+        return data
+
+
+    def instruments(self, symbol):
+        ''' Generates an instrument object. Currently this is only used for 
+        placing orders, and generating and using the instrument object are handled
+        for you, so you can ignore this method'''
+        res = self.session.get(self.endpoints['instruments'], params={'query':symbol.upper()})
+        if res.status_code == 200:
+            return res.json()['results']
+        else:
+            raise Exception("Could not generate instrument object: %s " % res.headers)
+
+
+    def quote_data(self, stock=''):
+        """Fetch stock quote
+
+            Args:
+                stock (str): stock ticker, prompt if blank
+
+            Returns:
+                (:obj:`dict`): JSON contents from `quotes` endpoint
+        """
+
+        url = None
+
+        if stock.find(',') == -1:
+            url = str(self.endpoints['quotes']) + str(stock) + "/"
+        else:
+            url = str(self.endpoints['quotes']) + "?symbols=" + str(stock)
+
+        #Check for validity of symbol
+        try:
+            req = requests.get(url)
+            req.raise_for_status()
+            data = req.json()
+        except requests.exceptions.HTTPError:
+            raise RH_exception.InvalidTickerSymbol()
```

Comment:
All HTTP errors will be treated as if we posted invalid symbol - which might not be the case. I'd suggest to do deeper inspection here. E.g. http 500, shall not raise InvalidTickerSymbol.